### ISSUE TITLE
Error on 1.20.6

### ISSUE THREAD

**Issue by mattheufarmer** [rosebudmods/rainglow#33]
I have been using this mod for a while with no issue, but this afternoon, for no reason I can figure out, it started causing me to disconnect from the server I play on with "Network Protocol Error" and I could not reconnect. I've been in contact with the owner and there are no errors reported on their side of things, but when I checked my logs, I found this:

```
[16:10:07] [Render thread/ERROR]: Failed to handle packet net.minecraft.class_8042@3439baf4
java.lang.IllegalStateException: Invalid entity data item type for field 16 on entity class_5776[' Glow Squid'/2010, l='ClientLevel', x=-7081.92, y=18.13, z=-13879.88]: old=blue(class java.lang.String), new=85(class java.lang.Integer)
	at net.minecraft.class_2945.method_12785(class_2945.java:126) ~[client-intermediary.jar:?]
	at net.minecraft.class_2945.method_12779(class_2945.java:117) ~[client-intermediary.jar:?]
	at net.minecraft.class_634.method_11093(class_634.java:550) ~[client-intermediary.jar:?]
	at net.minecraft.class_2739.method_11808(class_2739.java:54) ~[client-intermediary.jar:?]
	at net.minecraft.class_2739.method_11054(class_2739.java:12) ~[client-intermediary.jar:?]
	at net.minecraft.class_634.method_48294(class_634.java:2197) ~[client-intermediary.jar:?]
	at net.minecraft.class_8042.method_48329(class_8042.java:19) ~[client-intermediary.jar:?]
	at net.minecraft.class_8042.method_11054(class_8042.java:7) ~[client-intermediary.jar:?]
	at net.minecraft.class_2600.method_11072(class_2600.java:25) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_18859(class_1255.java:162) [client-intermediary.jar:?]
	at net.minecraft.class_4093.method_18859(class_4093.java:23) [client-intermediary.jar:?]
	at net.minecraft.class_1255.method_16075(class_1255.java:136) [client-intermediary.jar:?]
	at net.minecraft.class_1255.method_5383(class_1255.java:121) [client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1275) [client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:888) [client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:265) [client-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470) [fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.15.11.jar:?]
```

While that doesn't immediately point to this mod, it is the only mod I have that has anything to do with changing the color of glow squids, and once I disabled it, I was able to reconnect to the server with no issue.

During my search I noticed that another issue had mentioned something similar (#28), so whatever is going on here, might be linked to that - I don't know.

Let me know if you need any other information from me. Thanks for an awesome mod!

Edit: fixed formatting of the log

**Comment by ix0rai** [comment]
I haven't been able to reproduce this issue -- if you could create a minimal set of mods that need to be present before it occurs, that would be great!

**Comment by ix0rai** [comment]
@mattheufarmer any updates on this?

**Comment by A5ho9999** [comment]
I went to look into this a little, I believe this is most likely due to the networking and component/nbt changes in 1.20.5/6. As it stands at least with the 1.21 version I just tested the mod doesn't function correctly on the client side when on a server that does not have the mod installed. The colour changes aren't applied.

I was unable to reproduce it with just the 1.21 version but if I had to guess it's more likely caused by playing on a 1.20.4 or earlier server that doesn't have the mod which was then upgraded to 1.20.5/6. Could also be completely wrong but that's my best guess at the moment. I'll try to look into it more once I've completed my current project if still unable to reproduce.
